check_maintenance_incidents.py

```diff
@@ -29,6 +30,17 @@ def add_devel_project_review(self, req, package):
         if action_is_patchinfo(a):
             a = req.actions[1]
         project = a.tgt_releaseproject if a.type == 'maintenance_incident' else req.actions[0].tgt_project
+
+        if project.startswith('openSUSE:'):
+            prj, pkg = devel_project_get(self.apiurl, "openSUSE:Factory", package)
+            if prj is None and package.find(".") > 0:
+                prj, pkg = devel_project_get(self.apiurl, "openSUSE:Factory", package[0:package.rfind(".")])
+            logging.debug(f'using devel project {prj}/{pkg}')
+            if prj is not None:
```

Comment:
that block seems to be a duplication of lines https://github.com/openSUSE/openSUSE-release-tools/pull/3224/files#diff-877efb0a9116edf680121175667b645eabb70f056d9d4c91b641377a39269380R49-R54 

why are we not implementing this logic in owner_fallback (by e.g. changing it to return proj, package instead of the entire xml tree..

Reply:
```
check_maintenance_incidents.py:from osclib.core import owner_fallback
check_maintenance_incidents.py:        root = owner_fallback(self.apiurl, project, package)
osclib/core.py:def owner_fallback(apiurl, project, package):
osclib/core.py:    root = owner_fallback(apiurl, project, package)
```

The owner_fallback is basically only used here and it returns devel project and maintainer information, not not devel project info. Re-implementing OBS XML output here is not really right approach.

Reply:
And yes, it's duplication, because it's simplest solution without need to create additional complexity.

Reply:
I'd prefer if line 35-39 would be cacheable via memoize rather than repeated lookups. here's what we could do: 

* fold the existing owner_fallback implementation into `maintainers_get` (inline it). 
* change owner_fallback to just return a `project/package` tuple. 
* move line 35-39 into `owner_fallback`
*

Reply:
Then maybe we can rename owner_fallback to "find_devel_project"?